toolchain/check/pattern_match.cpp

```diff
@@ -320,7 +334,8 @@ auto MatchContext::DoEmitPatternMatch(Context& context,
            .pretty_name_id = SemIR::GetPrettyNameFromPatternId(
                context.sem_ir(), entry.pattern_id)});
       AddWork({.pattern_id = param_pattern.subpattern_id,
-               .scrutinee_id = param_id});
+               .scrutinee_id = param_id,
+               .ref_binding_context = false});
```

Comment:
This makes it look like `entry.ref_binding_context` is being overridden, but IIUC it's guaranteed to already be false, so it's effectively being propagated. I wonder if it would be clearer to instead have this

```suggestion
               .ref_binding_context = entry.ref_binding_context});
```

and maybe add `CARBON_CHECK(entry.ref_binding_context == false);` above. I'm of two minds because it's more code, and it trades off being a little less explicit about the actual value, so I'm OK either way, but I wanted to mention the option.

If you decide to go with this, I think it would also apply in `OutParamPattern` and `RefParamPattern` (with the `CHECK` inverted in the latter case).

Reply:
Done.